test/cpp/naming/gen_build_yaml.py

```diff
@@ -24,16 +24,6 @@
 
 _LOCAL_DNS_SERVER_ADDRESS = '127.0.0.1:15353'
 
-def _expect_target_is_balancer(test_case):
-  records = test_case['records']
-  srv_attempted = '_grpclb._tcp.%s' % test_case['record_to_resolve']
-  if records.get(srv_attempted) is None:
-    return False
-  for r in records[srv_attempted]:
-    if r['type'] == 'SRV':
-      return True
-  return False
-
 def _expected_chosen_service_config(test_case):
```

Comment:
Instead of having an `expected_config_index` attribute, I suggest having a `expected_service_config` attribute that contains the expected string.  This is because when the TXT record contains multiple choices, the resolver is responsible for picking one of those choices.  The service config it returns should include only the selected choice -- i.e., it's a subset of the TXT record.

For example, one of the tests you've defined has the following TXT record:

```
grpc_config=[
  {"percentage":0,
   "serviceConfig":{"loadBalancingPolicy":"round_robin","methodConfig":[{"name":[{"method":"Foo","service":"NeverPickedService","waitForReady":true}]}]}},
  {"percentage":100,
   "serviceConfig":{"loadBalancingPolicy":"round_robin","methodConfig":[{"name":[{"method":"Foo","service":"AlwaysPickedService","waitForReady":true}]}]}}]
```

In this case, the resolver should return only the second service config:

```
{"loadBalancingPolicy":"round_robin","methodConfig":[{"name":[{"method":"Foo","service":"AlwaysPickedService","waitForReady":true}]}]}}
```